### PR TITLE
Add cli option to Galley to allow metadata on outgoing sink connections.

### PR DESCRIPTION
**Pull request by jeffmendoza**
For use with sinkAddress, outgoing connections to MCP sink servers
will have gRPC stream metadata attached as defined by sinkMeta.

### CODE REVIEW

**Comment by ozevren** [comment]
Can we support foo=bar style parameters? That should help with the len(items)%2==0 problem as well.

**Comment by jeffmendoza** [comment]
Ah ok, I see StringToString now, will give it a shot.

https://godoc.org/github.com/spf13/pflag#FlagSet.StringToStringVar

File: galley/pkg/server/callout.go [hunk 1]
```diff
@@ -56,11 +59,13 @@ func defaultCalloutPT() calloutPatchTable {
 	}
 }
 
-func newCallout(address, auth string, so *source.Options) (*callout, error) {
-	return newCalloutPT(address, auth, so, defaultCalloutPT())
+func newCallout(address, auth string, meta []string,
```
Comment (ozevren) [inline galley/pkg/server/callout.go]: Nit: meta => metadata. Also some comments about its k,v,k,v... nature.

File: galley/pkg/server/args.go [hunk 2]
```diff
@@ -110,6 +114,7 @@ func DefaultArgs() *Args {
 		DomainSuffix:              defaultDomainSuffix,
 		DisableResourceReadyCheck: false,
 		ExcludedResourceKinds:     defaultExcludedResourceKinds(),
+		SinkMeta:                  make([]string, 0),
```
Comment (ozevren) [inline galley/pkg/server/args.go]: Add it to String() as well?

File: galley/pkg/server/callout_test.go [hunk 3]
```diff
@@ -19,27 +19,37 @@ import (
...
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
 
 	mcp "istio.io/api/mcp/v1alpha1"
 	"istio.io/istio/pkg/mcp/source"
 )
 
 func TestCallout(t *testing.T) {
-	co, err := newCallout("foo", "NONE", &source.Options{})
+	co, err := newCallout("foo", "NONE", []string{"foo=bar"}, &source.Options{})
```
Comment (ozevren) [inline galley/pkg/server/callout_test.go]: Test cases for a=b=c, "a=" "=b", "="?

**Comment by ozevren** [comment]
e2e-mixer-noauth-v1alpha3-v2 failure seems unrelated
```
2019-02-11T20:40:04.903963Z	error	Error starting port forwarder: failure running port forward process: error upgrading connection: error dialing backend: dial tcp: lookup default-3f86ccab-ef66-43ca-bbe8-78d78ad49872 on 169.254.169.254:53: dial udp 169.254.169.254:53: operation was canceled
```

**Comment by ozevren** [comment]
@wenchenglu this is ready to go

**Pull request merged**